src/dotnet/CarbonAware.WebApi/Configuration/CarbonAwareConfigurationExtension.cs

```diff
@@ -0,0 +1,11 @@
+using CarbonAware.Plugins.JsonReaderPlugin;
+
+namespace CarbonAware.WebApi.Configuration;
+
+public static class CarbonAwareServicesConfiguration
+{
+    public static void AddCarbonAwareServices(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.AddSingleton<ICarbonAware, CarbonAwareJsonReaderPlugin>();
```

Comment:
May want to use TryAddSingleton instead.  If this is called multiple times, you'll end up with MULTIPLE singletons in the DI container, which is probably not expected.

Reply:
I think we should be determining the plugin to attach to ICarbonAware based on configuration?  That's probably an integration thing.